### PR TITLE
Use node groups in all examples and documentation, deprecate top level node group variables in partition.

### PR DESCRIPTION
**Pull request by heyealex**
### Description
In this PR:
* Examples have been updated to use node groups
* Documentation has been updated to do the same
* Top level node group variables have been deprecated, if they are used an error will display pointing to the node group module.
* Default node group has been removed from partition.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/compute/schedmd-slurm-gcp-v5-node-group/README.md [hunk 1]
```diff
@@ -36,50 +36,6 @@ module as input with:
     partition_name: compute
 ```
 
-### Compute VM Zone Policies
```
Comment (tpdownes) [inline community/modules/compute/schedmd-slurm-gcp-v5-node-group/README.md:39]: I'm a little concerned that this documentation may be disappearing. Unclear if accidental or intentional due to #662

**Review by tpdownes — CHANGES_REQUESTED** [review]
Address question re: README section disappearing.

File: community/modules/compute/schedmd-slurm-gcp-v5-node-group/README.md (hunk 1 above)
Comment (heyealex) [reply community/modules/compute/schedmd-slurm-gcp-v5-node-group/README.md:39]: The same content exists already in community/modules/compute/schedmd-slurm-gcp-v5-partition/README.md, which is where zone policies are defined. 
It was mistakenly added in both places in a prior PR, and in retrospect it should have been mentioned in the description.

**Review by tpdownes — DISMISSED** [review]
I have not tested all the blueprint changes, but I will rely upon `make tests` and previous implementation work to presume success.

**Review by tpdownes — DISMISSED** [review]

**Review by tpdownes — APPROVED** [review]

**Pull request merged**